src/fuel-vm/instruction-set.md

```diff
@@ -1297,6 +1298,21 @@ Panic if:
 
 - `$pc + ($rC + imm + 1) * 4 > VM_MAX_RAM - 1`
 
+### `JAL`: Jump and link
+
+|             |                                                                                               |
+|-------------|-----------------------------------------------------------------------------------------------|
+| Description | Set `$rA` to address of the next instruction. Jump to instruction at address `$rB + imm * 4`. |
+| Operation   | `if $rA is not $zero { $rA = $pc + 4 }` <br> `$pc = $rB + imm * 4`                            |
+| Syntax      | `jal $rA $rB imm`                                                                             |
+| Encoding    | `0x00 rA rB i i`                                                                              |
+| Notes       | If `$rA` is `$zero`, the return address discarded.                                            |
+
+Panic if:
+
+- `$rA` is a reserved register other than `$zero`
+- `$rB + imm * 4 >= VM_MAX_RAM`
+
```

Comment:
Should we panic if $rB == $pc && imm == 0 to avoid jumping into the exact same spot?

Reply:
Likely not. It's not like you couldn't otherwise make an infinite loop if you want, and then you'll just run out of gas anyway.